src/lib/OTA/OTA.cpp

```diff
@@ -183,4 +195,96 @@ void ICACHE_RAM_ATTR UnpackChannelDataSeqSwitches(volatile uint8_t* Buffer, CRSF
             break;
     }
 }
+#endif
 #endif // SEQ_SWITCHES
+
+#if (!defined HYBRID_SWITCHES_8 and !defined SEQ_SWITCHES) or defined UNIT_TEST
+
+#if TARGET_TX or defined UNIT_TEST
+
+void ICACHE_RAM_ATTR GenerateChannelData10bit(volatile uint8_t* Buffer, CRSF *crsf, uint8_t addr)
+{
+  uint8_t PacketHeaderAddr;
+  PacketHeaderAddr = (addr << 2) | RC_DATA_PACKET;
+  Buffer[0] = PacketHeaderAddr;
+  Buffer[1] = ((crsf->ChannelDataIn[0]) >> 3);
+  Buffer[2] = ((crsf->ChannelDataIn[1]) >> 3);
+  Buffer[3] = ((crsf->ChannelDataIn[2]) >> 3);
+  Buffer[4] = ((crsf->ChannelDataIn[3]) >> 3);
+  Buffer[5] = ((crsf->ChannelDataIn[0] & 0b110) << 5) |
+                           ((crsf->ChannelDataIn[1] & 0b110) << 3) |
+                           ((crsf->ChannelDataIn[2] & 0b110) << 1) |
+                           ((crsf->ChannelDataIn[3] & 0b110) >> 1);
+#ifdef One_Bit_Switches
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[4]) << 7;
```

Comment:
This either needs one of the below to reset the byte to 0.  Other wise switches will not turn off.

Buffer[6] |= 0;

or

Buffer[6] = CRSF_to_BIT(crsf->ChannelDataIn[4]) << 7;

Reply:
good catch